rollup/internal/controller/watcher/chunk_proposer.go

```diff
@@ -196,52 +141,11 @@ func (p *ChunkProposer) TryProposeChunk() {
 }
 
 func (p *ChunkProposer) updateDBChunkInfo(chunk *encoding.Chunk, codecVersion encoding.CodecVersion, metrics *utils.ChunkMetrics) error {
-	if chunk == nil {
+	if chunk == nil || len(chunk.Blocks) == 0 {
 		return nil
 	}
 
-	compatibilityBreachOccurred := false
-
-	for {
```

Comment:
Why exactly is this check not needed anymore?

Reply:
added it back in [6a8ef2c](https://github.com/scroll-tech/scroll/pull/1658/commits/6a8ef2c3d387dd731d41941622b687a46f0e7789).